carbon-design-system/carbon
# feat(eslint): add eslint-plugin-jest

Closes #9047 

This PR adds in `eslint-plugin-jest` to our ESLint config. It also sets up a separate `internal.js` file so that we can use new rules without impacting the Public API until we're ready to make the change.

#### Changelog

**New**

- Add jest config and `eslint-plugin-jest` to our eslint config package

**Changed**

- Update tests to new eslint rules

**Removed**

#### Testing / Reviewing

- Verify the ESLint rules are being applied
- Verify ESLint passes as expected
- Verify tests pass as expected
